Gameplay/misc/LaserController.cpp

```diff
@@ -7,177 +7,186 @@
-	, _toggle_active_time(1.0f)
-	, _toggle_inactive_time(1.0f)
-{}
+    : Script(game_object, "LaserController")
+      , _laser(nullptr)
+      , _max_length(5.0f)
+      , _max_scale(0.2f)
+      , _activation_time(1.0f)
+      , _damage(1.0f)
+      , _spin_movement(false)
+      , _spin_speed(20.0f)
+      , _horizonal_movement(false)
+      , _movement_speed(1.0f)
+      , _movement_target(nullptr)
+      , _toggle_activation(false)
+      , _toggle_active_time(1.0f)
+      , _toggle_inactive_time(1.0f)
+{
+}
 
 void Hachiko::Scripting::LaserController::OnAwake()
 {
-	_terrain = Scenes::GetTerrainContainer();
-	_player = Scenes::GetPlayer();
+    _terrain = Scenes::GetTerrainContainer();
+    _player = Scenes::GetPlayer();
 
-	_initial_position = game_object->GetTransform()->GetGlobalPosition();
+    _initial_position = game_object->GetTransform()->GetGlobalPosition();
 }
 
 void Hachiko::Scripting::LaserController::OnUpdate()
 {
-	if (Input::IsKeyDown(Input::KeyCode::KEY_1))
-	{
-		if (State::ACTIVE) ChangeState(State::INACTIVE);
-		if (State::INACTIVE) ChangeState(State::ACTIVATING);
-	}
-
-	if (_spin_movement)
-	{
-		float3 euler_rotation = game_object->GetTransform()->GetLocalRotationEuler();
-		euler_rotation.y += (_spin_clockwise) ? Time::DeltaTime() * _spin_speed * -1.0f : Time::DeltaTime() * _spin_speed;
-		game_object->GetTransform()->SetLocalRotationEuler(euler_rotation);
-	}
-
-	if (_horizonal_movement && _movement_target != nullptr)
-	{
-		_movement_position += (_movement_forward_direction) ? _movement_speed * Time::DeltaTime() : _movement_speed * Time::DeltaTime() * -1.0f;
-
-		if (_movement_position > 1.0f)
-		{
-			_movement_forward_direction = false;
-			_movement_position += (1.0f - _movement_position);
-		}
-		else if (_movement_position < 0.0f)
-		{
-			_movement_forward_direction = true;
-			_movement_position *= -1.0f;
-		}
-
-		game_object->GetTransform()->SetGlobalPosition(
-			float3::Lerp(_initial_position, _movement_target->GetTransform()->GetGlobalPosition(), _movement_position * _movement_position * (3 - 2 * _movement_position))
-		);
-	}
-	
-	switch (_state)
-	{
-	case State::ACTIVE:
-		AdjustLength();
-		CheckPlayerCollision();
-		if (_toggle_activation) 
-		{
-			_elapsed_time += Time::DeltaTime();
-			if (_elapsed_time >= _toggle_active_time)
-			{
-				ChangeState(State::INACTIVE);
-			}
-		}
-		break;
-
-	case State::ACTIVATING:
-		if (!_laser->IsActive()) 
-		{
-			_laser->SetActive(true);
-		}
-
-		AdjustLength();
-		_elapsed_time += Time::DeltaTime();
-		_scale = (_elapsed_time / _activation_time) * _max_scale;
-		if (_elapsed_time >= _activation_time) 
-		{
-			ChangeState(State::ACTIVE);
-		}
-		break;
-
-	case State::INACTIVE:
-		if (_toggle_activation) 
-		{
-			_elapsed_time += Time::DeltaTime();
-			if (_elapsed_time >= _toggle_inactive_time)
-			{
-				ChangeState(State::ACTIVATING);
-			}
-		}
-		break;
-	}
+    if (IsKeyDown(Input::KeyCode::KEY_1))
+    {
+        if (_state == ACTIVE)
```

Comment:
@miquelmiro3 check this, previously it was only 
if (ACTIVE)
which would create unreachable code

Reply:
This was debug code thanks for pointing it out. It can be deleted.